as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/StandardRSEJBossStartLaunchDelegate.java

```diff
@@ -87,4 +169,58 @@ protected boolean isStarted(IServer server) {
 	protected void pollServer(IServer server, final boolean expectedState) {
 		PollThreadUtils.pollServer(server, expectedState);
 	}
+	
+	protected boolean setServerAlreadyStarted(ILaunchConfiguration configuration, String mode, 
+			IControllableServerBehavior beh) throws CoreException {
+		boolean ret = super.setServerAlreadyStarted(configuration, mode, beh);
+		connectDebugger(configuration, mode, beh);
+		return ret;
+	}
+	
+	protected void connectDebugger(ILaunchConfiguration configuration, String mode, 
+			IControllableServerBehavior beh) throws CoreException {
+		attachRemoteDebugger(beh.getServer(), getDebugPort(beh.getServer()), new NullProgressMonitor());
+	}
+	
+
+	private ILaunch attachRemoteDebugger(IServer server, int localDebugPort, IProgressMonitor monitor) 
+			throws CoreException {
+		monitor.subTask("Attaching remote debugger");
```

Comment:
User can't type a negative number into the UI. I've added another check that if it's < 1, to use the default value.